coroutineadapter/src/main/java/cz/ackee/retrofitadapter/chain/CallChain.kt

```diff
@@ -26,12 +25,10 @@ class CallChainImpl(
     override val annotations: Array<out Annotation>
         get() = annotationArray
 
-    override fun proceed(call: Call<*>): Deferred<*> {
-        if (chainIndex + 1 > interceptors.size) {
-            throw IllegalStateException("chainIndex ${chainIndex + 1} does not match with actual interceptor size ${interceptors.size}")
-        }
+    override fun proceed(call: Call<*>): Call<*> {
+        check(chainIndex + 1 > interceptors.size) {"chainIndex ${chainIndex + 1} does not match with actual interceptor size ${interceptors.size}" }
```

Comment:
I like the extension `lastIndex`, you dont have to perform any +1 magic.

Reply:
✅